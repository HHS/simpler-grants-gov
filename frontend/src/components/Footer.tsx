import GrantsLogo from "public/img/grants-logo.svg";
import { ExternalRoutes } from "src/constants/routes";
import { UswdsIconNames } from "src/types/generalTypes";

import { useTranslations } from "next-intl";
import Image from "next/image";
import {
  Address,
  Grid,
  GridContainer,
  SocialLinks,
  Footer as USWDSFooter,
} from "@trussworks/react-uswds";

import { USWDSIcon } from "src/components/USWDSIcon";

// Recreate @trussworks/react-uswds SocialLink component to accept any Icon
// https://github.com/trussworks/react-uswds/blob/cf5b4555e25f0e52fc8af66afe29253922bed2a5/src/components/Footer/SocialLinks/SocialLinks.tsx#L33
type SocialLinkProps = {
  href: string;
  name: string;
  icon: string;
};

const SocialLink = ({ href, name, icon }: SocialLinkProps) => (
  <a className="usa-social-link" href={href} title={name} target="_blank">
    <USWDSIcon
      className="usa-social-link__icon"
      height="40px"
      name={icon as UswdsIconNames}
      aria-label={name}
    />
  </a>
);

const Footer = () => {
  const t = useTranslations("Footer");

  const links = [
    {
      href: ExternalRoutes.GRANTS_X_TWITTER,
      name: t("link_x_twitter"),
      icon: "x",
    },
    {
      href: ExternalRoutes.GRANTS_YOUTUBE,
      name: t("link_youtube"),
      icon: "youtube",
    },
    {
      href: ExternalRoutes.GRANTS_BLOG,
      name: t("link_blog"),
      icon: "local_library",
    },
    {
      href: ExternalRoutes.GRANTS_NEWSLETTER,
      name: t("link_newsletter"),
      icon: "mail",
    },
    {
      href: ExternalRoutes.GRANTS_RSS,
      name: t("link_rss"),
      icon: "rss_feed",
    },
    {
      href: ExternalRoutes.GITHUB_REPO,
      name: t("link_github"),
      icon: "github",
    },
  ].map(({ href, name, icon }) => (
    <SocialLink href={href} key={name} name={name} icon={icon} />
  ));

  return (
    <USWDSFooter
      data-testid="footer"
      size="medium"
      returnToTop={
        <GridContainer className="usa-footer__return-to-top margin-top-5">
          <a href="#">{t("return_to_top")}</a>
        </GridContainer>
      }
      primary={
        <GridContainer>
          <Grid row gap>
            <Grid tablet={{ col: 4 }}>
              <div className="footer-logo-container position-relative dekstop:height-5">
                <Image
                  className="height-auto position-relative"
                  alt={t("logo_alt")}
                  src={GrantsLogo as string}
                  unoptimized
                  fill
                />
              </div>
            </Grid>
            <Grid className="usa-footer__contact-links" tablet={{ col: 8 }}>
              <SocialLinks links={links} />
              <h2 className="usa-footer__contact-heading text-balance">
                {t("agency_contact_center")}
              </h2>
              <Address
                size="medium"
                items={[
                  <a key="telephone" href={`tel:${t("telephone")}`}>
                    {t("telephone")}
                  </a>,
                  <a
                    key="email"
                    href={`mailto:${ExternalRoutes.EMAIL_SUPPORT}`}
                    target="_blank"
                  >
                    {ExternalRoutes.EMAIL_SUPPORT}
                  </a>,
                ]}
              />
            </Grid>
          </Grid>
        </GridContainer>
      }
      secondary={null}
    />
  );
};

export default Footer;
