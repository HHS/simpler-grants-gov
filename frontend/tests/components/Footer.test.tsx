import { render, screen } from "tests/react-utils";

import Footer from "src/components/Footer";

describe("Footer", () => {
  it("Renders without errors", () => {
    render(<Footer />);
    const footer = screen.getByTestId("footer");
    expect(footer).toBeInTheDocument();
  });
  it("matches snapshot", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
