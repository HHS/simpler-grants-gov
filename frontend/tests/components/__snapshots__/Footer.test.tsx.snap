// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<div>
  <footer
    class="usa-footer usa-footer--medium"
    data-testid="footer"
  >
    <div
      class="grid-container usa-footer__return-to-top margin-top-5"
      data-testid="gridContainer"
    >
      <a
        href="#"
      >
        Return to top
      </a>
    </div>
    <div
      class="usa-footer__primary-section"
    >
      <div
        class="grid-container"
        data-testid="gridContainer"
      >
        <div
          class="grid-row grid-gap"
          data-testid="grid"
        >
          <div
            class="tablet:grid-col-3"
            data-testid="grid"
          >
            <div
              class="footer-logo-container position-relative"
            >
              <img
                alt="Grants.gov logo"
                class="height-auto position-relative"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                src="/img.jpg"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
              />
            </div>
          </div>
          <div
            class="tablet:grid-col-3 usa-footer__nav border-0"
            data-testid="grid"
          >
            <h3>
              Explore
            </h3>
            <ul
              class="margin-top-3"
            >
              <li
                class="usa-footer__secondary-link margin-left-0"
              >
                <a
                  href="/"
                >
                  Home
                </a>
              </li>
              <li
                class="usa-footer__secondary-link margin-left-0"
              >
                <a
                  href="/{to}"
                >
                  Search
                </a>
              </li>
              <li
                class="usa-footer__secondary-link margin-left-0"
              >
                <a
                  href="/{to}"
                >
                  Vision
                </a>
              </li>
              <li
                class="usa-footer__secondary-link margin-left-0"
              >
                <a
                  href="/{to}"
                >
                  Roadmap
                </a>
              </li>
              <li
                class="usa-footer__secondary-link margin-left-0"
              >
                <a
                  href="/{to}"
                >
                  Events
                </a>
              </li>
              <li
                class="usa-footer__secondary-link margin-left-0"
              >
                <a
                  href="/{to}"
                >
                  Newsletter
                </a>
              </li>
            </ul>
          </div>
          <div
            class="tablet:grid-col-3 margin-top-3 tablet:margin-top-0"
            data-testid="grid"
          >
            <h3>
              Simpler.Grants.gov
            </h3>
            <div>
              To give feedback, contact: 
              <a
                href="mailto:simpler@grants.gov"
              >
                simpler@grants.gov
              </a>
            </div>
            <h3>
              Grants.gov Support Center
            </h3>
            <div>
              For technical support, contact: 
              <a
                href="mailto:support@grants.gov"
              >
                support@grants.gov
              </a>
            </div>
            <div
              class="margin-top-2"
            >
              1-800-518-4726
            </div>
          </div>
          <div
            class="tablet:grid-col-3 margin-top-3 tablet:margin-top-0"
            data-testid="grid"
          >
            <h3
              class=""
            >
              Grants.gov Program Management Office
            </h3>
            <div>
              Grantors, contact the PMO through your 
              <a
                href="https://www.grants.gov/grantors/grantor-help"
                rel="noopener noreferrer"
                target="_blank"
              >
                Agency Point of Contact
              </a>
              .
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="usa-footer__secondary-section"
    >
      <div
        class="grid-container"
      />
    </div>
  </footer>
</div>
`;
